crates/wast/src/core/binary.rs

```diff
@@ -212,6 +212,11 @@ impl Encode for Type<'_> {
 
 impl Encode for Rec<'_> {
     fn encode(&self, e: &mut Vec<u8>) {
+        if self.types.len() == 1 {
+            self.types[0].encode(e);
+            return;
+        }
```

Comment:
Was this required to get tests working, and if so which one? Otherwise I'd prefer to not do this to avoid possibly-surprising changes to the AST of the binary emitted

Reply:
This was needed to get round-tripping of `types-subtyping.wast` working, I'll add the `module binary` thing you suggested below.

Reply:
Which test needed this? Is it something that should be changed upstream?